coldfront/core/utils/management/commands/audit_data.py

```diff
@@ -0,0 +1,229 @@
+                    ' is inactive and has unexpired allocations.' \
+                        .format(project.id, project.name)))
+                try:
+                    allocation_attribute = allocation.allocationattribute_set.get(allocation_attribute_type__name='Service Units')
+                    if allocation_attribute.value == 0:
+                        self.stdout.write(self.style.ERROR('Project {} ({})'
+                    ' is inactive and has non-zero SUs.' \
+                            .format(project.id, project.name)))
+                except AllocationAttribute.DoesNotExist as e:
+                    pass
+
+    def handle_project_pi(self):
+        '''
+        Assert that all projects have at least one PI.
+        '''
+        projects = Project.objects.all()
+        for project in projects:
+            if project.pis().count() == 0:
+                self.stdout.write(self.style.ERROR('Project {} ({})'
+                    ' has no PIs.' \
+                    .format(project.id, project.name)))
+
+    def handle_user_project(self):
+        '''
+        Assert that all users with a cluster UID should be associated with at least one Project.
+        '''
+        users = UserProfile.objects.filter(cluster_uid__isnull=False) \
+            .values('id', 'user__first_name', 'user__last_name', 'user__email', 'access_agreement_signed_date')
+        for user in users:
+            user_project_count = ProjectUser.objects.filter(user_id=user['id']).count()
+            if user_project_count == 0:
+                self.stdout.write(self.style.ERROR('User {} ({} {}, {})'
+                ' has a cluster UID but is not associated with any projects.{}'
+                    .format(user['id'], user['user__first_name'],
+                    user['user__last_name'], user['user__email'],
+                    '' if user['access_agreement_signed_date'] \
+                        else ' They have not signed the access agreement.')))
+
+    def handle_lrc_user_billing(self):
+        '''
+        Assert that all LRC users have a billing_activity.
+        '''
+        users = UserProfile.objects.filter(cluster_uid__isnull=False) \
+            .values('id', 'billing_activity', 'user__first_name',
+                    'user__last_name', 'user__email')
+        for user in users:
+            if user['billing_activity'] is None:
+                self.stdout.write(self.style.ERROR('User {} ({} {}, {})'
+                ' has a cluster UID but is not associated with any billing activity.' \
+                    .format(user['id'], user['user__first_name'],
+                    user['user__last_name'], user['user__email'])))
+
+    def handle_lrc_recharge_allocation_billing(self):
+        '''
+        Assert that all LRC Recharge Allocations have a "Billing Activity"-typed AllocationAttribute.
+        '''
+        allocations = Allocation.objects.filter(project__cluster__name='Lawrencium')
```

Comment:
project__cluster__name is not a valid lookup. I think we need to filter on the Resource having a 'Compute' name since there's a lot of clusters on the LRC side

Reply:
Oh thanks. I stopped working on the LRC stuff because I don't know if I'm only supposed to work on BRC and also I'm not familiar with LRC resources and allocations. I'll change the filter and add a `TODO` for every LRC command in a while.